Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -260,5 +257,21 @@ private void RemoveSecurityFromUniverse(Universe universe, Security member, List
             // remove symbol mappings for symbols removed from universes // TODO : THIS IS BAD!
             SymbolCache.TryRemove(member.Symbol);
         }
+
+        /// <summary>
+        /// Determines if we can safely remove the security member from a universe.
+        /// We must ensure that we have zero holdings, no open orders, and no existing portfolio targets
+        /// </summary>
+        private bool IsSafeToRemove(Security member)
+        {
+            // but don't physically remove it from the algorithm if we hold stock or have open orders against it or an open target
+            var openOrders = _algorithm.Transactions.GetOrders(x => x.Status.IsOpen() && x.Symbol == member.Symbol);
+            if (!member.HoldStock && !openOrders.Any() && (member.Holdings.Target == null || member.Holdings.Target.Quantity == 0))
```

Comment:
Future me: `!member.HoldStock .... member.Holdings.Target.Quantity == 0` no holdings and not aiming for any more.